grizli/prep.py

```diff
@@ -3271,26 +3410,33 @@ def process_direct_grism_visit(direct={},
             file = '{0}_wcs.{1}'.format(direct['product'], ext)
             if os.path.exists(file):
                 os.remove(file)
+        # I think I need to cast all of the values in the dq extension to be numpy.int16 
+        #(as opposed to numpy.uint32 which they apparently currently are)
+        
 
         # First drizzle
         if len(direct['files']) > 1:
+            for file in direct['files']:
+                hdu = pyfits.open(file,mode='update')
+                hdu[3].data = hdu[3].data.astype(np.int16)
+                hdu.flush()
             AstroDrizzle(direct['files'], output=direct['product'],
-                         clean=True, context=False, preserve=False,
+                         clean=True, context=False, preserve=False, 
                          skysub=True, driz_separate=True, driz_sep_wcs=True,
                          median=True, blot=True, driz_cr=True,
                          driz_cr_snr=driz_cr_snr_first,
                          driz_cr_scale=driz_cr_scale_first,
                          driz_cr_corr=False, driz_combine=True,
                          final_bits=bits, coeffs=True, build=False,
-                         final_wht_type='IVM', **drizzle_params)
+                         final_wht_type='IVM', gain='-99', rdnoise='-99', **drizzle_params)
```

Comment:
we might need to be careful with these.  Also the "ERR" extensions of JWST seem to be split into different components for read noise and Poisson.